src/services/expense.service.js

```diff
@@ -0,0 +1,80 @@
+const { Op } = require('sequelize');
+const { Expense } = require('../models/Expense.model.js');
+
+const getExpenses = async ({ userId, categoryList, from, to }) => {
+  const whereClause = {};
+
+  if (userId) {
+    whereClause.userId = userId;
+  }
+
+  if (categoryList && categoryList.length > 0) {
+    whereClause.category = { [Op.in]: categoryList };
+  }
+
+  if (from) {
+    whereClause.spentAt = { [Op.gte]: new Date(from) };
+  }
+
+  if (to) {
+    whereClause.spentAt = { ...whereClause.spentAt, [Op.lte]: new Date(to) };
```

Comment:
Similar to the previous comment, ensure that the `to` date is valid before using it to create a new Date object. Consider adding validation to handle invalid date formats.